src/System.Private.CoreLib/shared/System/AppDomain.cs

```diff
@@ -131,7 +131,7 @@ private int ExecuteAssembly(Assembly assembly, string?[]? args)
                 parameters: entry.GetParameters().Length > 0 ? new object?[] { args } : null,
                 culture: null);
 
-            return result != null ? (int)result : 0;
+            return (int?) result ?? 0;
```

Comment:
These are not the same; the latter produces IL that involves nullable.  Please revert this and all similar changes.